src/messaging/ExchangeContext.h

```diff
@@ -183,6 +183,7 @@ class DLL_EXPORT ExchangeContext : public ReferenceCounted<ExchangeContext, Exch
                             ExchangeDelegate * delegate);
     void Free();
     void Reset();
+    void ForceFree();
```

Comment:
The semantics of this need to be very clearly documented.  After this is called is it safe (in the "won't crash" sort of sense) to call SendMessage?   Will the exchange context be reused for anything until the refcount actually drops to 0 (sounds like no?)?  That sort of thing.

Are we basically doing the equilent of Abort() but without dropping the refcount here?  It looks to me like the differences are:

* ForceFree does not clear the retrans table.
* ForceFree does not flush pending acks
* ForceFree does not decrement the refcount
* ForceFree _does_ do the work Free() does.

One thing that confuses me: if we ForceFree and then later the consumer drops the ref so that we land in Free(), we're going to crash because `mExchangeMgr` is null so the VerifyOrDie will fail.  Is that the expected behavior?

We also end up with a kinda-weird value of mContextsInUse on the `ExchangeManager`, but looks like that's only used for DETAIL_LOGGING anyway (raising the question of why we maintain it all the time).

It would _amost_ make sense to me to have a `ForceFree` which basically does `DoClose(true)` followed by `DoFree()` which does the work of `Free` except for the refcount assert.  And change `Free` to silently bail out on null `mExchangeMgr`, since that means we were force-freed.  It would still be a kinda weird state, but....

Except I guess we don't want to flush acks like `DoClose(true)` does?  But I'd think we _do_ want to clear the retrans table?

Reply:
The problem this PR trying to solve was raised from a CI fail. A exchange context is leaked, and its timer is triggered after the exchange manager has been released, then a invalid memory access is triggered in the timer handler. Because the crash is located inside the timer handler, and there are very little information in the stack, it is very hard to track the true cause.

The PR is providing a very hacky workaround to release the timer. But after reading your comments, I realized that this is not a good solution, I'd better crash the process when a exchange context is leaked.